src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/authorization/ServiceProviderAccessScopeMapper.kt

```diff
@@ -83,8 +86,9 @@ class ServiceProviderAccessScopeMapper(
       emptyList()
     } else {
       val contracts = dynamicFrameworkContractRepository.findAllByContractReferenceIn(contractGroups)
-      if (contracts.isEmpty()) {
-        configErrors.add("user has no valid contract groups configured")
+      val removedContracts = contractGroups.subtract(contracts.map(DynamicFrameworkContract::contractReference))
```

Comment:
i've not seen this type of argument to 'map' before - does it do the same as 

`val removedContracts = contractGroups.subtract(contracts.map { it.contractReference })`?

Reply:
yes, it's the method reference syntax: https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html

> 

>     Arrays.sort(rosterAsArray,
>         (a, b) -> Person.compareByAge(a, b)
>     );
>
> Because this lambda expression invokes an existing method, you can use a method reference instead of a lambda expression:
>
>     Arrays.sort(rosterAsArray, Person::compareByAge);

Reply:
In hindsight, it's probably not worth the extra characters...